Zend/zend_inheritance.c

```diff
@@ -619,9 +619,7 @@ static void do_inheritance_check_on_method(zend_function *child, zend_function *
 		zend_string *method_prototype = zend_get_function_declaration(parent);
 		zend_string *child_prototype = zend_get_function_declaration(child);
 
-		if (child->common.prototype && (
-			child->common.prototype->common.fn_flags & ZEND_ACC_ABSTRACT
-		)) {
+		if (parent->common.fn_flags & ZEND_ACC_ABSTRACT) {
```

Comment:
If B extends A implements I and B violates I then you *should* get a fatal. The whole goal is to eventually fatal for all incompatibilties. The only reason I can see to not give a fatal is if we previously didn't fatal, in which case it should probably wait until 8.0.